lib/dry/struct/class_interface.rb

```diff
@@ -52,16 +52,15 @@ def check_schema_duplication(new_schema)
       private :check_schema_duplication
 
       def constructor_type(type = nil)
-        if type
-          @constructor_type = type
-        else
-          @constructor_type || :strict
-        end
+        return @constructor_type = type if type
+        @constructor_type
       end
 
       def schema
+        @schema = {} unless defined? @schema
```

Comment:
Ok. I meant like after we specify `constructor_type` we could just set the class instance variable